azurerm/resource_arm_storage_account.go

```diff
@@ -909,6 +917,11 @@ func expandStorageAccountNetworkRules(d *schema.ResourceData) *storage.NetworkRu
 	networkRule := networkRules[0].(map[string]interface{})
 	networkRuleSet := &storage.NetworkRuleSet{}
 
+	if networkRule["default_action"].(string) == string(storage.DefaultActionAllow) {
+		// If Default Access is enabled then no network rules are set.
```

Comment:
Is this required? will the API be displeased if you still provide them?

Reply:
API is not displeased if we provide network rules.  However, I am not able to override those IP Rules/Virtual Network Rules even after specifying empty list. 

```
if networkRule["default_action"].(string) == string(storage.DefaultActionAllow) {
		// If Default Access is enabled then no network rules are set.
		return &storage.NetworkRuleSet{
			DefaultAction:       storage.DefaultActionAllow,
			Bypass:              storage.Bypass(""),
			IPRules:             &[]storage.IPRule{},
			VirtualNetworkRules: &[]storage.VirtualNetworkRule{},
		}
	}
```
Also, API creates random key value pair which is hard to determine. Eg:

```
          network_rules.0.bypass.625562768 = AzureServices
          network_rules.0.default_action = Allow
          network_rules.0.ip_rules.# = 1
          network_rules.0.ip_rules.3619153832 = 127.0.0.1
```
Hence, I cant unset `625562768` or `3619153832` .

Reply:
Ahh so if you specify a default action azure auto populates that property with some values, could you potentially mark it as computed if that is the case?

Reply:
Done. Marked variables as computed. Also added test to reflect the changes.

Reply:
You are still returning an object with just the default action here.

What i was getting at is not returning an object here, setting DefaultAction and all the other properties regardless of what default action is.